iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -621,24 +727,37 @@ private async Task RefreshSasTokenLoopAsync(DateTime refreshesOn, CancellationTo
 
         private async Task VerifyIsOpenAsync(CancellationToken cancellationToken)
         {
-            await _handlerSemaphore.WaitAsync(cancellationToken);
+            ThrowIfDisposed();
+            await _clientOpenSemaphore.WaitAsync(cancellationToken);
 
             try
             {
-                if (!_isOpen)
+                if (GetClientTransportStatus() != ClientTransportStatus.Open)
```

Comment:
This will result in operations invoked when the client is "opening" to also get rejected. It seemed like a reasonable expectation since we do want the client behavior to be deterministic.